uptane/clients/primary.py

```diff
@@ -459,58 +455,63 @@ def get_validated_target_info(self, target_filepath):
           'to allow some targets to validate without Director approval, or is'
           'the wrong repository specified as the Director repository in the '
           'initialization of this primary object?')
-    
-    director_target = validated_target_info[self.director_repo_name]
 
+    director_target = validated_target_info[self.director_repo_name]
+    hardware_id_release_counter = set() # To check if the hardwareIDs and release counters are matching in all the repos other than director
+    director_hardware_id_release_counter = set() # To check with that of the other repos
```

Comment:
I don't see the changes requested here:
- not distinguishing between Director and Image Repository in the code here
- using literal string and int rather than sets whose length you check

[Here's the full version of the resulting primary.py, with the section of code missing discussed changes highlighted](https://github.com/shikhar394/uptane/blob/86b826c7cdedf0b2c58ed9671a8984335f4b8cbe/uptane/clients/primary.py#L420-L484) at the time of this comment.